x-pack/plugins/infra/public/hooks/use_log_view.ts

```diff
@@ -5,118 +5,126 @@
  * 2.0.
  */
 
+import { useInterpret, useSelector } from '@xstate/react';
 import createContainer from 'constate';
-import { useCallback, useEffect, useState } from 'react';
-import type { HttpHandler } from '@kbn/core/public';
-import { LogView, LogViewAttributes, LogViewStatus, ResolvedLogView } from '../../common/log_views';
+import { useCallback, useState } from 'react';
+import { waitFor } from 'xstate/lib/waitFor';
+import { LogViewAttributes } from '../../common/log_views';
+import {
+  createLogViewNotificationChannel,
+  createLogViewStateMachine,
+} from '../observability_logs/log_view_state';
 import type { ILogViewsClient } from '../services/log_views';
-import { isRejectedPromiseState, useTrackedPromise } from '../utils/use_tracked_promise';
+import { isDevMode } from '../utils/dev_mode';
 
 export const useLogView = ({
   logViewId,
   logViews,
-  fetch,
+  useDevTools = isDevMode(),
 }: {
   logViewId: string;
   logViews: ILogViewsClient;
-  fetch: HttpHandler;
+  useDevTools?: boolean;
 }) => {
-  const [logView, setLogView] = useState<LogView | undefined>(undefined);
-
-  const [resolvedLogView, setResolvedLogView] = useState<ResolvedLogView | undefined>(undefined);
-
-  const [logViewStatus, setLogViewStatus] = useState<LogViewStatus | undefined>(undefined);
-
-  const [loadLogViewRequest, loadLogView] = useTrackedPromise(
+  const [logViewStateNotifications] = useState(() => createLogViewNotificationChannel());
```

Comment:
**nit**: if this is the case whether `logViewStateNotifications` is not a reactive value (it never triggers a rerender), using a `useRef` call may be more suitable for this use case:
```js
const logViewStateNotificationsRef = useRef(createLogViewNotificationChannel())
```
If the `createLogViewNotificationChannel` is an expensive computation and we want to still lazy load the initial value that we'll never update again, we also have two possible approaches using `useRef` or `useMemo`:
```js
const logViewStateNotificationsRef = useRef(null)
if (logViewStateNotificationsRef.current === null) {
  logViewStateNotificationsRef.current = createLogViewNotificationChannel()
}

// or with useMemo
const logViewStateNotifications = useMemo(createLogViewNotificationChannel, [])
```
The expected result is identical to using `useState`, the only difference is we stop treating this value as a reactive one.

Reply:
> if the `createLogViewNotificationChannel` is an expensive computation and we want to still lazy load the initial value that we'll never update again, we also have two possible approaches using `useRef` or `useMemo`

It creates an observable, several functions, and objects so I would say it's too expensive to unnecessarily throw away on each render.

Overall, can you explain why you don't agree with this use of `useState`? The single instance of the notification channel is "state" with the special property that we currently don't have the need to replace it.

The semantics of `useRef` and `useMemo` actually fit worse IMHO, because

- `useRef` creates a container for a value that can change over time without the identity of the container changing. This is not what we need here.
- `useMemo` doesn't make any guarantees about when and how often the value is calculated. The docs explicitly state that one should not rely on that for referential stability.

So `useState` with an initializer, but without a `setState` is a really good match to implement a value whose identity is tied to the lifetime of the component.

Does that make sense?